crates/re_ui/examples/re_ui_example.rs

```diff
@@ -600,3 +645,788 @@ impl egui_tiles::Behavior<Tab> for MyTileTreeBehavior {
+// DRAG AND DROP DEMO
+
+mod drag_and_drop {
+    use std::collections::HashSet;
+
+    #[derive(Hash, Clone, Copy, PartialEq, Eq)]
+    struct ItemId(u32);
+
+    pub struct ExampleDragAndDrop {
+        items: Vec<(ItemId, String)>,
+
+        /// currently selected items
+        selected_items: HashSet<ItemId>,
+    }
+
+    impl Default for ExampleDragAndDrop {
+        fn default() -> Self {
+            Self {
+                items: (0..10).map(|i| (ItemId(i), format!("Item {i}"))).collect(),
+                selected_items: HashSet::new(),
+            }
+        }
+    }
+
+    impl ExampleDragAndDrop {
+        pub fn ui(&mut self, re_ui: &crate::ReUi, ui: &mut egui::Ui) {
+            let mut source_item_pos = None;
+            let mut target_item_pos = None;
+
+            for (i, (item_id, label)) in self.items.iter_mut().enumerate() {
+                //
+                // Draw the item
+                //
+
+                let id = egui::Id::new("drag_demo").with(*item_id);
+
+                let response = re_ui
+                    .list_item(label.as_str())
+                    .selected(self.selected_items.contains(item_id))
+                    .drag_id(id)
+                    .show(ui);
+
+                //
+                // Handle item selection
+                //
+
+                // Basic click and cmd/ctr-click
+                if response.clicked() {
+                    if ui.input(|i| i.modifiers.command) {
+                        if self.selected_items.contains(item_id) {
+                            self.selected_items.remove(item_id);
+                        } else {
+                            self.selected_items.insert(*item_id);
+                        }
+                    } else {
+                        self.selected_items.clear();
+                        self.selected_items.insert(*item_id);
+                    }
+                }
+
+                // Multi-selection dragging not (yet?) supported, so dragging resets selection to single item.
+                // TODO(emilk/egui#3841): it would be nice to have response.decidedly_dragged()
+                if response.dragged() {
+                    // Here, we support dragging a single item at a time, so we set the selection to the dragged item
+                    // if/when we're dragging it proper.
+                    if ui.input(|i| i.pointer.is_decidedly_dragging()) {
+                        self.selected_items.clear();
+                        self.selected_items.insert(*item_id);
+                    }
+                }
+
+                //
+                // Detect end-of-drag situation and prepare the swap command.
+                //
+
+                if response.dragged() || response.drag_released() {
+                    source_item_pos = Some(i);
+                }
+
+                // TODO(emilk/egui#3841): this feels like a common enough pattern that is should deserve its own API.
```

Comment:
https://github.com/emilk/egui/issues/3841 was a narrow issue about how `hovered` should work.

I think you should link to a tracking issue about improved drag-and-drop support instead, and collect your thoughts there